### PR TITLE
Fix another JMS readonly exception

### PR DESCRIPTION
**Pull request by SylvainJuge**
## What does this PR do?

Improves previous fix #1715 as there was another direct writing to JMS message not covered.

## Checklist

- [x] This is a bugfix
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/master/CHANGELOG.asciidoc)
  - [ ] I have added tests that would fail without this fix

### CODE REVIEW

File: apm-agent-plugins/apm-jms-plugin/apm-jms-plugin-base/src/main/java/co/elastic/apm/agent/jms/JmsInstrumentationHelper.java [hunk 1]
```diff
@@ -129,39 +131,35 @@ public Span startJmsSendSpan(Destination destination, Message message) {
...
-                span.getContext().getDestination().getService()
-                    .withName("jms")
-                    .withResource("jms")
-                    .withType(MESSAGING_TYPE);
-                if (destinationName != null) {
-                    span.getContext().getDestination().getService().getResource().append("/").append(destinationName);
-                    span.withName("JMS SEND to ");
-                    addDestinationDetails(null, destination, destinationName, span);
-                    if (isDestinationNameComputed) {
-                        message.setStringProperty(JMS_DESTINATION_NAME_PROPERTY, destinationName);
```
Comment (SylvainJuge) [inline apm-agent-plugins/apm-jms-plugin/apm-jms-plugin-base/src/main/java/co/elastic/apm/agent/jms/JmsInstrumentationHelper.java:144]: the offending writing call was right here.

File: apm-agent-plugins/apm-jms-plugin/apm-jms-plugin-base/src/main/java/co/elastic/apm/agent/jms/JmsMessagePropertyAccessor.java [hunk 2]
```diff
@@ -83,9 +83,9 @@ public void setHeader(String headerName, String headerValue, Message message) {
         try {
             message.setStringProperty(headerName, headerValue);
         } catch (MessageNotWriteableException e) {
-            logger.debug("Failed to set JMS message property due to read-only message", e);
+            logger.debug("Failed to set JMS message property {} due to read-only message", headerName, e);
         } catch (JMSException e) {
-            logger.warn("Failed to set JMS message property. Distributed tracing cannot work without that.");
+            logger.warn("Failed to set JMS message property {}.", headerName);
```
Comment (SylvainJuge) [inline apm-agent-plugins/apm-jms-plugin/apm-jms-plugin-base/src/main/java/co/elastic/apm/agent/jms/JmsMessagePropertyAccessor.java:88]: here the name of the property will make it obvious to know the case and if it's related to distributed tracing or not.
Now this is also used to add the destination name for the receiver side.

File: apm-agent-plugins/apm-jms-plugin/apm-jms-plugin-base/src/main/java/co/elastic/apm/agent/jms/JmsInstrumentationHelper.java [hunk 3]
```diff
@@ -49,6 +49,8 @@
 
 public class JmsInstrumentationHelper {
 
+    private static final JmsMessagePropertyAccessor MSG_ACCESSOR = JmsMessagePropertyAccessor.instance();
```
Comment (eyalkoren) [inline apm-agent-plugins/apm-jms-plugin/apm-jms-plugin-base/src/main/java/co/elastic/apm/agent/jms/JmsInstrumentationHelper.java:52]: [minor] It's already a singleton, I don't think it makes sense to hold a reference to it

File: apm-agent-plugins/apm-jms-plugin/apm-jms-plugin-base/src/main/java/co/elastic/apm/agent/jms/JmsMessagePropertyAccessor.java (hunk 2 above)
Comment (eyalkoren) [reply apm-agent-plugins/apm-jms-plugin/apm-jms-plugin-base/src/main/java/co/elastic/apm/agent/jms/JmsMessagePropertyAccessor.java:88]: It's not for us, it's for the user who doesn't necessarily knows this property is related to distributed tracing. In addition, this message is still valid - if we can't set a property, distributed tracing will not work. We can say "Distributed tracing _**may**_ not work..." instead.

**Review by eyalkoren — APPROVED** [review]

**Pull request merged**